webpack/assets/javascripts/react_app/components/MemoryAllocationInput/index.js

```diff
@@ -1,3 +1,41 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { setMemoryValue } from './MemoryAllocationInputActions';
+
+import reducer from './MemoryAllocationInputReducer';
+
 import MemoryAllocationInput from './MemoryAllocationInput';
 
-export default MemoryAllocationInput;
+// export reducers
+export const reducers = { memoryAllocationInput: reducer };
+
+const ConnectedMemoryAllocationInput = props => {
+  const dispatch = useDispatch();
+  const setMemoryAction = value => dispatch(setMemoryValue(value));
+
+  const memoryValue = useSelector(state => state.memoryAllocationInput.value);
+
+  useEffect(() => {
```

Comment:
This updated the memory value when `defaultValue` has been changed, though I'm not sure if I understand the reason for this correlation between the default value and the actual value.

Reply:
The default value is the initialized value but for example, in ovirt, it can be changed according to template, cluster, or instance type so the default value changes as well.
Let me know if I haven't understood your question correctly